hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/MongoSqlConnectorBase.java

```diff
@@ -0,0 +1,248 @@
+
+import com.hazelcast.function.FunctionEx;
+import com.hazelcast.jet.core.DAG;
+import com.hazelcast.jet.core.EventTimePolicy;
+import com.hazelcast.jet.core.Vertex;
+import com.hazelcast.jet.sql.impl.connector.SqlConnector;
+import com.hazelcast.jet.sql.impl.connector.SqlProcessors;
+import com.hazelcast.spi.impl.NodeEngine;
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.expression.ExpressionEvalContext;
+import com.hazelcast.sql.impl.row.JetSqlRow;
+import com.hazelcast.sql.impl.schema.ConstantTableStatistics;
+import com.hazelcast.sql.impl.schema.MappingField;
+import com.hazelcast.sql.impl.schema.Table;
+import com.hazelcast.sql.impl.schema.TableField;
+import org.apache.calcite.rex.RexNode;
+import org.bson.BsonDocument;
+import org.bson.conversions.Bson;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.hazelcast.jet.core.Edge.between;
+import static com.hazelcast.jet.mongodb.impl.Mappers.defaultCodecRegistry;
+import static com.hazelcast.sql.impl.type.QueryDataType.VARCHAR;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Base for MongoDB SQL Connectors.
+ *
+ * Streaming and batch connectors have similar way of dealing with scans with few exceptions (like the requirement
+ * for {@code _id field}).
+ *
+ * All MongoDB connectors assume one primary key: {@code _id} column, which is mandatory (auto-created if not specified
+ * by user), unique and indexed.
+ *
+ * <p>
+ * While secondary indexes are technically possible, they are not supported right now.
+ *
+ * @see FieldResolver
+ */
+public abstract class MongoSqlConnectorBase implements SqlConnector {
+
+    private final FieldResolver fieldResolver = new FieldResolver();
+
+    @Nonnull
+    @Override
+    public List<MappingField> resolveAndValidateFields(
+            @Nonnull NodeEngine nodeEngine,
+            @Nonnull Map<String, String> options,
+            @Nonnull List<MappingField> userFields,
+            @Nonnull String externalName
+    ) {
+        return fieldResolver.resolveFields(options, userFields, isStream());
+    }
+
+    @Nonnull
+    @Override
+    public List<String> getPrimaryKey(Table table) {
+        MongoTable mongoTable = (MongoTable) table;
+        return mongoTable.primaryKeyName();
+    }
+
+    @Nonnull
+    @Override
+    public Table createTable(@Nonnull NodeEngine nodeEngine, @Nonnull String schemaName, @Nonnull String mappingName,
+                             @Nonnull String externalName, @Nonnull Map<String, String> options,
+                             @Nonnull List<MappingField> resolvedFields) {
+        String databaseName = options.get(Options.DATABASE_NAME_OPTION);
+        String collectionName = options.get(Options.COLLECTION_NAME_OPTION);
```

Comment:
Seems that we use a specific option to specify the external table name (the collection). We should use `externalName` for this.

Reply:
Done